pinot-broker/src/test/java/com/linkedin/pinot/broker/routing/builder/KafkaHighLevelConsumerRoutingTableBuilderTest.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.broker.routing.builder;
+
+import com.linkedin.pinot.broker.routing.RoutingTableLookupRequest;
+import com.linkedin.pinot.common.config.TableConfig;
+import com.linkedin.pinot.common.utils.CommonConstants;
+import com.linkedin.pinot.common.utils.HLCSegmentName;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import org.apache.commons.configuration.BaseConfiguration;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.InstanceConfig;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class KafkaHighLevelConsumerRoutingTableBuilderTest {
+
+  public static final String ALL_PARTITIONS = "ALL";
+
+  @Test
+  public void testHlcRoutingTableBuilder() {
+    final int ITERATIONS = 50;
+    final int MAX_INSTANCES_PER_GROUP = 5;
+    final int MAX_NUM_GROUPS = 5;
+    final int MAX_NUM_SEGMENTS = 100;
+
+    Random random = new Random();
+
+    KafkaHighLevelConsumerBasedRoutingTableBuilder routingTableBuilder =
+        new KafkaHighLevelConsumerBasedRoutingTableBuilder();
+    routingTableBuilder.init(new BaseConfiguration(), new TableConfig(), null, null);
+
+    String tableName = "table_REALTIME";
```

Comment:
It'd be better to rename it to `tableNameWithType`.

Reply:
done